src/Helper/Product/Utility.php

```diff
@@ -94,38 +91,47 @@ class Utility
     protected $filter;
     /** @var CoreInterface */
     protected $sgCore;
+    /** @var var GetStockIdForCurrentWebsite */
+    protected $getStockIdForCurrentWebsite;
+    /** @var GetStockItemDataInterface */
+    private $getStockItemData;
+    /** @var GetStockItemConfiguration */
+    protected $getStockItemConfiguration;
 
     /**
-     * @param StockItemResource           $stockItemResource
-     * @param StockItemInterfaceFactory   $stockItemFactory
      * @param StoreManager                $storeManager
      * @param TaxCalculation              $taxCalculation
      * @param ExportUtility               $utility
      * @param CategoryRepositoryInterface $categoryRepository
      * @param TaxConfig                   $taxConfig
      * @param FilterProvider              $filter
      * @param CoreInterface               $sgCore
+     * @param GetStockIdForCurrentWebsite $getStockIdForCurrentWebsite
+     * @param GetStockItemConfiguration   $getStockItemConfiguration
+     * @param GetStockItemDataInterface   $getStockItemData
      */
     public function __construct(
-        StockItemResource $stockItemResource,
-        StockItemInterfaceFactory $stockItemFactory,
         StoreManager $storeManager,
         TaxCalculation $taxCalculation,
         ExportUtility $utility,
         CategoryRepositoryInterface $categoryRepository,
         TaxConfig $taxConfig,
         FilterProvider $filter,
-        CoreInterface $sgCore
+        CoreInterface $sgCore,
+        GetStockIdForCurrentWebsite $getStockIdForCurrentWebsite,
```

Comment:
just move this stock stuff to a separate helper class, it will be better as this one has grown a bit too big

Reply:
not done

Reply:
ha, i think u misunderstood this point, I meant put all Stock management stuff into one class :D Utility is becoming too big and has too many responsibilities.